### PR TITLE
Added treemap example

### PR DESCRIPTION
**Pull request by justinpolygon**
This PR adds a tools directory for holding larger examples. The first one added is for a treemap visualization tool that integrates with Polygon.io + D3.js to display market movements.

Changes include:

- New `treemap_server.py` script to launch the visualization server.
- Utility script `polygon_sic_code_data_gatherer.py` to fetch and process SIC code data.
- Pre-computed SIC code mappings in `sic_code_groups.json`.
- Updated readme.md with instructions to set up and run the tool.

To test, run the `python3 treemap_server.py` script and navigate to http://localhost:8889 in a browser.

global

### CODE REVIEW

File: examples/tools/treemap/readme.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Mapping Market Movements with Polygon.io and D3.js Treemap
+
+This repository contains code and resources for visualizing stock market data using D3.js Treemap and Polygon.io's API.
+
```
Comment (mmoghaddam385) [inline examples/tools/treemap/readme.md:4]: Are you planning on linking the future blog post here? If so, will it be going into more detail about what this treemap represents (ie what does each box represent, what do the colors mean)? If the README explains exactly what we're making it makes reading and understanding the code easier.

I'd also suggest putting the screenshot from the PR in this README to make it more clear what this code does without having to clone and run it.

File: examples/tools/treemap/readme.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+## Structure
+
+The repo consists of:
+
+- `polygon_sic_code_data_gatherer.py`: Gathers data using Polygon's APIs and groups them by SIC codes.
```
Comment (mmoghaddam385) [inline examples/tools/treemap/readme.md:9]: What kind of "data"? We should clarify that this script generates the `sip_code_groups.json` file

File: examples/tools/treemap/readme.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+- `sic_code_groups.json`: Pre-built JSON file containing grouped ticker to SIC code data.
+- `treemap_server.py`: Simple server to host the treemap visualization.
```
Comment (mmoghaddam385) [inline examples/tools/treemap/readme.md:11]: We should clarify that this server relies on the `sic_code_groups.json` file

File: examples/tools/treemap/readme.md (hunk 1 above)
Comment (justinpolygon) [reply examples/tools/treemap/readme.md]: Fixed - added more an intro paragraph, a link to the tutorial, and an image.

File: examples/tools/treemap/readme.md (hunk 2 above)
Comment (justinpolygon) [reply examples/tools/treemap/readme.md]: Fixed -- updated the wording here to explain what's going on.

**Review by mmoghaddam385 — APPROVED** [review]

**Pull request merged**